src/OpenLoco/src/Graphics/SoftwareDrawingEngine.cpp

```diff
@@ -154,11 +154,16 @@ namespace OpenLoco::Gfx
             return;
         }
 
-        if (scaleFactor > 1.0f)
+        if (scaleFactor != 1.0f)
         {
-            const auto scale = std::ceil(scaleFactor);
             // We only need this texture when we have a scale above 1x, this texture uses the actual canvas size.
-            SDL_SetHint(SDL_HINT_RENDER_SCALE_QUALITY, "linear");
+            int scale = 1;
+            if (scaleFactor < 2.0f)
```

Comment:
Instead of hardcoding you should just specify it in formula i.e. `if (fabs(log2(scale) - log2(scaleFactor)) > 0.0001)` set hint. 

We want to set the hint on anything that isn't a power of 2.

Reply:
I'm confused why we would want linear interpolation for scale factors greater than 2.

Testing on v24.10, the interpolation doesn't appear to have any effect at scales 2.25, 2.5, 2.75, or 3.0. It does have visible effect at 3.25, but at that point it doesn't improve readability of text, and if anything I think it looks a little ugly.